applications/RANSApplication/custom_strategies/algebraic_flux_corrected_scalar_steady_scheme.h

```diff
@@ -0,0 +1,567 @@
+    }
+
+    AlgebraicFluxCorrectedScalarSteadyScheme(const double RelaxationFactor,
+                                             const Flags rBoundaryFlags,
+                                             const Variable<int>& rPeriodicIdVar)
+        : BaseType(),
+          mRelaxationFactor(RelaxationFactor),
+          mBoundaryFlags(rBoundaryFlags),
+          mrPeriodicIdVar(rPeriodicIdVar)
+    {
+        KRATOS_INFO("AlgebraicFluxCorrectedScalarSteadyScheme")
+            << " Using periodic residual based algebraic flux corrected scheme "
+               "with relaxation "
+               "factor = "
+            << std::scientific << mRelaxationFactor << "\n";
+    }
+
+    ~AlgebraicFluxCorrectedScalarSteadyScheme() override = default;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    void Initialize(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY
+
+        BaseType::Initialize(rModelPart);
+
+        if (mrPeriodicIdVar != Variable<int>::StaticObject())
+        {
+            const int number_of_conditions = rModelPart.NumberOfConditions();
+#pragma omp parallel for
+            for (int i_cond = 0; i_cond < number_of_conditions; ++i_cond)
+            {
+                const ModelPart::ConditionType& r_condition =
+                    *(rModelPart.ConditionsBegin() + i_cond);
+                if (r_condition.Is(PERIODIC))
+                {
+                    // this only supports 2 noded periodic conditions
+                    KRATOS_ERROR_IF(r_condition.GetGeometry().PointsNumber() != 2)
+                        << this->Info() << " only supports two noded periodic conditions. Found "
+                        << r_condition.Info() << " with "
+                        << r_condition.GetGeometry().PointsNumber() << " nodes.\n";
+
+                    const ModelPart::NodeType& r_node_0 = r_condition.GetGeometry()[0];
+                    const std::size_t r_node_0_pair_id =
+                        r_node_0.FastGetSolutionStepValue(mrPeriodicIdVar);
+
+                    const ModelPart::NodeType& r_node_1 = r_condition.GetGeometry()[1];
+                    const std::size_t r_node_1_pair_id =
+                        r_node_1.FastGetSolutionStepValue(mrPeriodicIdVar);
+
+                    KRATOS_ERROR_IF(r_node_0_pair_id != r_node_1.Id())
+                        << "Periodic condition pair id mismatch in "
+                        << mrPeriodicIdVar.Name() << ". [ " << r_node_0_pair_id
+                        << " != " << r_node_1.Id() << " ].\n";
+
+                    KRATOS_ERROR_IF(r_node_1_pair_id != r_node_0.Id())
+                        << "Periodic condition pair id mismatch in "
+                        << mrPeriodicIdVar.Name() << ". [ " << r_node_1_pair_id
+                        << " != " << r_node_0.Id() << " ].\n";
+                }
+            }
+        }
+
+        // Allocate auxiliary memory.
+        const auto num_threads = OpenMPUtils::GetNumThreads();
```

Comment:
I guess that at this point it is not possible to use the `ParallelUtils` in here...